WorkloadManagementSystem/Executor/JobScheduling.py

```diff
@@ -222,16 +222,12 @@ def optimizeJob( self, jid, jobState ):
     return self.__setJobSite( jobState, stageSites )
 
   def __applySiteFilter( self, sites, active = False, banned = False ):
-    filtered = list( sites )
-    if active:
-      for site in sites:
-        if site not in active:
-          filtered.remove( site )
-    if banned:
-      for site in banned:
-        if site in filtered:
-          filtered.remove( site )
-    return filtered
+    filtered = set( sites )
+    if active and isinstance( active, ( list, set, dict ) ):
+      filtered &= set( active )
+    if banned and isinstance( banned, ( list, set, dict ) ):
+      filtered -= set( banned )
+    return list( filtered )
 
```

Comment:
This is not fully correct. In fact, it wasn't correct even before.

I made what is the correct fixes in https://github.com/fstagni/DIRAC/commit/02f90336e6bd2626338fbec2f80309b6ecdfa9a8
where I also added the unit tests.